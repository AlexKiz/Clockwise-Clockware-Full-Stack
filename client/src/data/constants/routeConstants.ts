/* eslint-disable max-len */
/* eslint-disable no-unused-vars */
export enum URL {
    CITY = '/city',
    LOGIN = '/login',
    MASTER = '/master',
    USER = '/user',
    CLOCK = '/clocks',
    ORDER = '/order',
    AVAILABLE_MASTER = '/availableMasters',
    CITY_FOR_ORDER = '/cityForOrder',
    ORDER_FOR_RATE = '/orderForRate',
	ORDER_FOR_UPDATE = '/orderForUpdate',
	CITY_FOR_UPDATE = '/cityForUpdate',
    RATED_ORDER = '/ratedOrder',
    ORDERS_RATING = '/ordersRating',
	REGISTRATION = '/user/registration',
	VERIFY = '/user/verify',
	COMPLETE_ORDER = '/completeOrder',
	EXPORT_XLSX = '/exportXLSX',
	STRIPE = 'stripe',
	TOTAL_ORDERS_CHART = 'totalOrdersChart',
	TOTAL_ORDERS_CITIES_PIE_CHART = '/totalOrdersForCitiesPieChart',
    TOTAL_ORDERS_MASTERS_PIE_CHART = '/totalOrdersForMastersPieChart',
	MASTERS_STATISTICS_TABLE = '/mastersStatisticsTable',
	ORDERS_FOR_CALENDAR = '/ordersForCalendar',
}

export enum RESOURCE {
    ADMIN = 'admin',
	MASTER = 'master',
	CLIENT = 'client',
    CITY_CREATE = 'city-create',
    CITIES_LIST = 'cities-list',
    ORDER_CREATE = 'order-create',
    ORDERS_LIST = 'orders-list',
    MASTER_CREATE = 'master-create',
    MASTERS_LIST = 'masters-list',
    USER_CREATE = 'user-create',
    USERS_LIST = 'users-list',
    LOGIN = 'login',
    RATE = 'rate',
    REGISTRATION = 'registration-form',
	VERIFICATION = 'email-verification',
	STATISTICS = 'statistics',
	CALENDAR = 'calendar',
}

export enum PARAM {
    MASTER_ID = 'masterIdParam',
    MASTER_NAME = 'masterNameParam',
    USER_ID = 'userIdParam',
    USER_NAME = 'userNameParam',
    USER_EMAIL = 'userEmailParam',
    CITY_ID = 'cityIdParam',
    CITY_NAME = 'cityNameParam',
    ORDER_ID = 'orderIdParam',
    ORDER_DATE = 'orderDateParam',
    ORDER_TIME = 'orderTimeParam',
    CLOCK_ID = 'clockIdParam',
    RATING_ID = 'ratingIdentificatorParam',
	HASH_VERIFY = 'hashVerify',
	GENERATED = 'generated'
}

export const ROUTES = {
	ORDER_FORM: {
		path: '/',
		exact: true,
	},
	REGISTRATION: {
		path: `/${RESOURCE.REGISTRATION}`,
	},
	ADMIN_MASTERS_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.MASTERS_LIST}`,
	},
	ADMIN_MASTER_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.MASTER_CREATE}/:${PARAM.MASTER_ID}?/:${PARAM.MASTER_NAME}?`,
	},
	ADMIN_USERS_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.USERS_LIST}`,
	},
	ADMIN_USER_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.USER_CREATE}/:${PARAM.USER_ID}?/:${PARAM.USER_NAME}?/:${PARAM.USER_EMAIL}?`,
	},
	ADMIN_CITIES_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.CITIES_LIST}`,
	},
	ADMIN_CITY_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.CITY_CREATE}/:${PARAM.CITY_NAME}?`,
	},
	ADMIN_ORDERS_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.ORDERS_LIST}`,
	},
	ADMIN_STATISTICS: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.STATISTICS}`,
	},
	MASTER_ORDERS_LIST: {
		path: `/${RESOURCE.MASTER}/${RESOURCE.ORDERS_LIST}`,
	},
	MASTER_CALENDAR: {
		path: `/${RESOURCE.MASTER}/${RESOURCE.CALENDAR}`,
	},
	CLIENT_ORDERS_LIST: {
		path: `/${RESOURCE.CLIENT}/${RESOURCE.ORDERS_LIST}`,
	},
	ADMIN_ORDER_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.ORDER_CREATE}/:${PARAM.ORDER_ID}?`,
	},
	LOGIN: {
		path: `/${RESOURCE.LOGIN}`,
	},
	RATE_ORDER: {
		path: `/${RESOURCE.RATE}/:${PARAM.RATING_ID}?`,
	},
	USER_VERIFY: {
		path: `/${RESOURCE.VERIFICATION}/:${PARAM.HASH_VERIFY}?/:${PARAM.GENERATED}?`,
	},
};
