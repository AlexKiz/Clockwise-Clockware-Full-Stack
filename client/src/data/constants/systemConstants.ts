/* eslint-disable max-len */
export const OPENING_HOURS: string[] = ['09:00', '10:00', '11:00', '12:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00'];
export const ACCESS_TOKEN = 'accessToken';

export const VALID = {
	NAME: new RegExp(/^([(A-Za-zА-Яа-я]{3,49})$|^([A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50})$/),
	EMAIL: new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/),
	CITY: new RegExp(/^[A-Za-zА-Яа-я]{3,100}$|^[A-Za-zА-Яа-я]{3,49}[-\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	PASSWORD: new RegExp(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&]).{0,}$/),
};

export enum ROLE {
	ADMIN = 'admin',
    MASTER = 'master',
	CLIENT = 'client',
}

export enum SORTED_FIELD {
	ID = 'id',
	NAME = 'name',
	EMAIL = 'email',
	RATING = 'rating',
	CLOCK_SIZE = 'clock.size',
	USER_NAME = 'user.name',
	USER_EMAIL = 'user.email',
	CITY_NAME = 'city.name',
	MASTER_NAME = 'master.name',
	START_WORK_ON = 'startWorkOn',
	END_WORK_ON = 'endWorkOn',
	SMALL_CLOCKS = 'smallClocks',
	MEDIUM_CLOCKS = 'mediumClocks',
	LARGE_CLOCKS = 'largeClocks',
	COMPLETED = 'completed',
	UNCOMPLETED = 'uncompleted',
	EARNED_AMOUNT = 'earnedAmount'
}

export enum SORTING_ORDER {
	ASC = 'asc',
	DESC = 'desc'
}
