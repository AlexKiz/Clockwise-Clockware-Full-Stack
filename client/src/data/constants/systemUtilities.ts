/* eslint-disable require-jsdoc */
import {useCallback, useRef} from 'react';

export function debouncer(callback, delay: number) {
	const timer = useRef<null | NodeJS.Timer>(null);

	const debouncedCallback = useCallback((...args) => {
		if (timer.current) {
			clearTimeout(timer.current);
		}

		timer.current = setTimeout(() => {
			callback(...args);
		}, delay);
	}, [callback, delay]);

	return debouncedCallback;
}
