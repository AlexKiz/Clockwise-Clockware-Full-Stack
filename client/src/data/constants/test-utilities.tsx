import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter, Route} from 'react-router-dom';


export const renderWithRouter = ({component, path, exact = false}) => (
	render(
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>,
	)
);

export const rerenderWrapper = ({component, path, exact = false}) => {
	return (
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>
	);
};

export * from '@testing-library/react';
