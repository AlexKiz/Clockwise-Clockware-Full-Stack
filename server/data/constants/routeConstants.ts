/* eslint-disable no-unused-vars */
export enum URL {
    CITY = 'city',
    LOGIN = 'login',
    MASTER = 'master',
    USER = 'user',
    CLOCKS = 'clocks',
    ORDER = 'order',
    AVAILABLE_MASTER = 'availableMasters',
    CITY_FOR_ORDER = 'cityForOrder',
    ORDER_FOR_RATE = 'orderForRate',
    RATED_ORDER = 'ratedOrder',
    API = 'api'
}

export enum RESOURCE {
    RATE = 'rate'
}
