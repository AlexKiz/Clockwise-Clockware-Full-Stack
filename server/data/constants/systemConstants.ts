/* eslint-disable max-len */
export const VALID = {
	CITY_NAME: new RegExp(/^[A-Za-zА-Яа-я]{3,100}$|^[A-Za-zА-Яа-я]{3,49}[-\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	MASTER_NAME: new RegExp(/^[A-Za-zА-Яа-я]{3,49}$|^[A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	USER_NAME: new RegExp(/^[A-Za-zА-Яа-я]{3,49}$|^[A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	USER_EMAIL: new RegExp(/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/),
	DATE: new RegExp(/(\d{4}[-](0[1-9]|1[0-2])[-]([0-2]{1}\d{1}|3[0-1]{1})|([0-2]{1}\d{1}|3[0-1]{1})[-](0[1-9]|1[0-2])[-]\d{4})[T]([0-2]{1}\d{1}[:][0-5]{1}\d{1})[:]([0-5]{1}\d{1})[.]([0-5]{1}\d{2}[Z])/),
};

export const METHOD = 'OPTIONS';
