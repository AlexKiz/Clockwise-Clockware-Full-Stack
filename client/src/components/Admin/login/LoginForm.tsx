import axios from 'axios';
import {RESOURCE, URL} from '../../../data/constants/routeConstants';
import React, {useEffect, useState, FC} from 'react';
import {useHistory} from 'react-router-dom';
import PublicHeader from '../../Headers/PublicHeader';
import '../login/login-form.css';
import {LoginFormProps} from './componentConstants';
import {ACCESSTOKEN} from 'src/data/constants/systemConstants';

const LoginForm:FC<LoginFormProps> = () => {
	const history = useHistory();

	const [adminLogin, setAdminLogin] = useState<string>('');
	const [adminPassword, setAdminPassword] = useState<string>('');

	useEffect(() => {
		if (localStorage.getItem(ACCESSTOKEN)) {
			history.push(`/${RESOURCE.ADMIN}/${RESOURCE.ORDERS_LIST}`);
		}
	}, []);


	const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();
		const payload = {
			adminLogin,
			adminPassword,
		};

		try {
			const {headers: {authorization: accessToken}} = await axios.post(URL.LOGIN, payload);
			localStorage.setItem(ACCESSTOKEN, accessToken.split(' ')[1]);
			history.push(`/${RESOURCE.ADMIN}/${RESOURCE.ORDERS_LIST}`);
		} catch (e) {
			alert('Incorrect logging data');
			setAdminPassword('');
		}
	};

	return (
		<div>
			<PublicHeader/>

			<div className='container-form'>
				<form className='form' onSubmit={onSubmit}>
					<div>
						<div className='form-section'>
							<div className='form-input__label'>
								<label>Enter Admin Login:</label>
							</div>
							<input
								placeholder='Email'
								type='email'
								name='login'
								value={adminLogin}
								onChange = {(adminLoginEvent) => setAdminLogin(adminLoginEvent.target.value)}
								required
							>
							</input>
						</div>
						<div className='form-section'>
							<div className='form-input__label'>
								<label>Enter Admin Password:</label>
							</div>
							<input
								placeholder='Password'
								type='password'
								name='password'
								value={adminPassword}
								onChange = {(adminPasswordEvent) => setAdminPassword(adminPasswordEvent.target.value)}
							>
							</input>
						</div>
						<div className='form-button'>
							<button type='submit'>Sign In</button>
						</div>
					</div>
				</form>
			</div>

		</div>
	);
};

export default LoginForm;
