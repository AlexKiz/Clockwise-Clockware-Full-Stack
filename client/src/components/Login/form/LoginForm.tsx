import axios from 'axios';
import {URL} from '../../../data/constants/routeConstants';
import React, {useState, useEffect, FC} from 'react';
import {useHistory, useParams} from 'react-router-dom';
import PublicHeader from '../../Headers/PublicHeader';
import classes from './login-form.module.css';
import {LoginFormProps, roleMappingLoginPaths, validate} from './componentConstants';
import {ACCESS_TOKEN} from 'src/data/constants/systemConstants';
import {useFormik} from 'formik';
import {
	Button,
	FilledInput,
	FormControl,
	IconButton,
	InputAdornment,
	InputLabel,
	Stack,
	TextField,
	Typography,
	CircularProgress,
} from '@mui/material';
import {Visibility, VisibilityOff} from '@mui/icons-material';
import AlertMessage from 'src/components/Notification/AlertMessage';
import jwtDecode from 'jwt-decode';
import {BearerParser} from 'bearer-token-parser';
import {useTranslation} from 'react-i18next';
import {AlertNotification, Params} from 'src/data/types/types';


const LoginForm:FC<LoginFormProps> = () => {
	const history = useHistory();

	const {t} = useTranslation();
	const {paymentMessage} = useParams<Params>();
	const [showPassword, setShowPassword] = useState<boolean>(false);
	const [alertOptions, setAlertOptions] = useState<AlertNotification>({
		message: '',
		type: 'success',
		notify: false,
	});
	const [loading, setLoading] = useState<boolean>(false);

	const isOpen = (value:boolean) => {
		setAlertOptions({...alertOptions, notify: value});
	};

	const formik = useFormik({
		initialValues: {
			login: '',
			password: '',
		},
		validate,
		onSubmit: async (values) => {
			try {
				setLoading(true);
				const login = await axios.post(URL.LOGIN, {
					login: values.login,
					password: values.password,
				});
				const token = BearerParser.parseBearerToken(login.headers);
				localStorage.setItem(ACCESS_TOKEN, token);

				history.push(roleMappingLoginPaths[login.data.role]);
			} catch (e) {
				setLoading(false);
				setAlertOptions({
					message: 'Incorrect logging data',
					type: 'error',
					notify: true,
				});
			}
			formik.resetForm();
		},
	});


	useEffect(() => {
		if (paymentMessage === 'success') {
			setAlertOptions({
				message: 'Your order has been created! Please rate the master afterwards!',
				type: 'success',
				notify: true,
			});
		} else if (paymentMessage === 'error') {
			setAlertOptions({
				message: 'Something went wrong while payment',
				type: 'error',
				notify: true,
			});
		}
	}, []);


	useEffect(() => {
		const token = localStorage.getItem(ACCESS_TOKEN);
		if (token) {
			const {role} = jwtDecode<{role: string}>(token);

			history.push(roleMappingLoginPaths[role]);
		}
	}, []);


	return (
		<div>
			<PublicHeader/>
			<div className={classes.conteiner}>
				<div className={classes.container_form}>
					<form className={classes.form} onSubmit={formik.handleSubmit}>
						<Stack direction="column" justifyContent="center" spacing={1}>
							<div className={classes.form_section}>
								<div className={classes.form_input__label}>
									<Typography
										variant="h6"
										gutterBottom
										component="label"
									>
										{t('login.userLogin')}
									</Typography>
								</div>
								<TextField
									id="login"
									name='login'
									label={t('labels.email')}
									placeholder="example@mail.com"
									variant="filled"
									size="small"
									margin="dense"
									fullWidth
									value={formik.values.login}
									onChange={formik.handleChange}
									onBlur={formik.handleBlur}
									error={formik.touched.login && Boolean(formik.errors.login)}
									helperText={formik.touched.login && formik.errors.login}
									required
								/>
							</div>
							<div className={classes.form_section}>
								<div className={classes.form_input__label}>
									<Typography
										variant="h6"
										gutterBottom
										component="label"
									>
										{t('login.userPassword')}
									</Typography>
								</div>
								<FormControl fullWidth variant="filled">
									<InputLabel>{t('labels.password')}</InputLabel>
									<FilledInput
										id="filled-adornment-password"
										type={showPassword ? 'text' : 'password'}
										value={formik.values.password}
										onChange={formik.handleChange('password')}
										endAdornment={
											<InputAdornment position="end">
												<IconButton
													aria-label="toggle password visibility"
													onClick={() => setShowPassword((showPassword) => !showPassword)}
													edge="end"
												>
													{showPassword ? <VisibilityOff /> : <Visibility />}
												</IconButton>
											</InputAdornment>
										}
									/>
								</FormControl>
							</div>
							<div className={classes.form_section}>
								<Button
									variant="contained"
									type="submit"
									color='success'
									className={classes.form_btn}
									style={ {fontSize: 18, borderRadius: 15} }
									disabled={loading}
								>
									{t('buttons.signIn')}
									{loading && <CircularProgress
										size={56}
										color="success"
										sx={{
											position: 'absolute',
											top: '50%',
											left: '50%',
											marginTop: '-28px',
											marginLeft: '-28px',
										}}
									/>}
								</Button>
							</div>
						</Stack>
					</form>
				</div>
				{
					alertOptions.notify &&
					<AlertMessage
						alertType={alertOptions.type}
						message={alertOptions.message}
						isOpen={isOpen}
						notify={alertOptions.notify}
					/>
				}
			</div>
		</div>
	);
};

export default LoginForm;
