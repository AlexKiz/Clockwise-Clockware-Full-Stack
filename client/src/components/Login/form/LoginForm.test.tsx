import React from 'react';
import LoginForm from './LoginForm';
import {act, cleanup, fireEvent} from '@testing-library/react';
import {renderWithRouter, rerenderWrapper} from '../../../data/constants/test-utilities';

describe('Login Form Component', () => {
	beforeEach(() => {
		cleanup();
	});
	jest.mock('react-router-dom', () => ({
		...jest.requireActual('react-router-dom'),
		useHistory: () => ({
			push: jest.fn(),
		}),
		useLocation: () => ({
			pathname: '/login',
		}),
	}));

	it('matches snapshot', () => {
		const {asFragment} = renderWithRouter({
			component: <LoginForm />,
			path: '/login',
		});

		expect(asFragment()).toMatchSnapshot();
	});

	it('shows inputs with empty values', () => {
		const {getByTestId} = renderWithRouter({
			component: <LoginForm />,
			path: '/login',
		});

		expect((getByTestId('login-email-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('login-password-input') as HTMLInputElement).value).toBe('');
	});

	it('triggers event handlers on inputs', () => {
		const {getByTestId, rerender} = renderWithRouter({
			component: <LoginForm />,
			path: '/login',
		});

		act(() => {
			fireEvent.change(getByTestId('login-email-input'), {
				target: {value: 'testLogin@gmail.com'},
			});
			fireEvent.change(getByTestId('login-password-input'), {
				target: {value: 'TestPassword1!'},
			});

			rerender(rerenderWrapper({
				component: <LoginForm />,
				path: '/login',
			}));
		});

		expect((getByTestId('login-email-input') as HTMLInputElement).value).toBe('testLogin@gmail.com');
		expect((getByTestId('login-password-input') as HTMLInputElement).value).toBe('TestPassword1!');
	});
});
