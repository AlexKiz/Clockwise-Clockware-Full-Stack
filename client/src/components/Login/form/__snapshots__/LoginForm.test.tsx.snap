// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <div
        class="wrapper_header"
      >
        <div
          class="wrapper_logo"
        >
          <a
            href="/"
          >
            <div
              class="inner_logo"
            >
              <div
                class="inner_logo_img"
              >
                <div
                  class="logo_img1a"
                >
                  <div
                    class="logo_img1b"
                  >
                    <div
                      class="logo_img1c"
                    />
                  </div>
                </div>
              </div>
              <div
                class="logo_text1d"
              />
            </div>
          </a>
        </div>
        <nav>
          <ul
            class="nav__links"
          >
            <li
              class=""
            >
              <a
                href="/"
              >
                Make order
              </a>
            </li>
            <li
              class="active"
            >
              <a
                href="/login"
              >
                Log in
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/registration-form"
              >
                Sign up
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1s4qybw-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  role="button"
                  tabindex="0"
                >
                  English
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="en"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      class="conteiner"
    >
      <div
        class="container_form"
      >
        <form
          class="form"
        >
          <div
            class="css-1nw2civ-MuiStack-root"
          >
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Enter User Login:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="login"
                  id="login-label"
                >
                  Email
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-10i04qz-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="login-email-input"
                    id="login"
                    name="login"
                    placeholder="example@mail.com"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Enter User Password:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                >
                  Password
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-f6v4u4-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="login-password-input"
                    id="filled-adornment-password"
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="VisibilityIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="form_section"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form_btn css-ke5b6m-MuiButtonBase-root-MuiButton-root"
                style="font-size: 18px; border-radius: 15px;"
                tabindex="0"
                type="submit"
              >
                Sign In
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
